android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt

```diff
@@ -270,4 +242,35 @@ fun hasPermission(context: Context, permission: String): Boolean {
 /** Returns true if the application has the given feature.  */
 fun hasFeature(context: Context, feature: String): Boolean {
     return context.packageManager.hasSystemFeature(feature)
-}
\ No newline at end of file
+}
+
+
+/**
+ * Workaround for not able to get device id on Android 10 or above using DRM API
+ * {@see https://stackoverflow.com/questions/58103580/android-10-imei-no-longer-available-on-api-29-looking-for-alternatives}
+ * {@see https://developer.android.com/training/articles/user-data-ids}
+ */
+fun getUniqueID(): String? {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2)
```

Comment:
given that we support api 16+ we should have some fallback logic here

Reply:
good point. the `getDeviceId` method does a null check and returns random uuid as fallback.